tests/sles4sap/wizard_hana_install.pm

```diff
@@ -42,26 +43,42 @@ sub try_reclaiming_space {
         $root_free *= 1024;
     }
     elsif ($units ne 'G') {
-        # Asume there's not enough free space on /dev/system/root if units are not T or G
+        # Assume there's not enough free space on /dev/system/root if units are not T or G
         $root_free = 0;
     }
+
     # Always leave at least 25Gb free on /dev/system/root for packages and Hana
     $root_free -= 25;
     $root_size -= $root_free if ($root_free > 0);
     if ($root_size >= ($root_used + 25)) {
-        assert_script_run "btrfs filesystem resize $root_size$units /";
-        assert_script_run "lvreduce --yes --force --size $root_size$units /dev/system/root";
+        assert_script_run "lvresize --yes --force --resizefs --size -$root_size$units /dev/system/root";
     }
     $output = script_output q@pvscan | sed -n '/system/s/\[//p' | awk '(NSIZE=$6-$9+1) {print $2","NSIZE","$10}'@;
     my ($device, $newsize, $unit) = split(/,/, $output);
     $unit = substr($unit, 0, 1);
+
     # Do nothing else unless there's at least 1GB to reclaim in the LVM partition
     return unless ($unit eq 'G' or $unit eq 'T');
+
+    # Set unit as GB, because adding TB is too much!
+    if ($unit eq 'T') {
+        $newsize *= 1024;
+        $unit = 'G';
+    }
+
+    # We can have allocated space at the end of the PV, so we need to move it before
+    my $script = 'reclaim_free_space_in_pv.sh';
+    assert_script_run "curl -f -v " . autoinst_url . "/data/sles4sap/$script -o /tmp/$script; chmod +x /tmp/$script";
+    assert_script_run "$script $device";
```

Comment:
Shouldn't we run the script with `set -e`? I would either add a `bash -e` in this line, or better yet a `set -e` in the script itself.

Reply:
Done!

In fact I'm usually don't use `set -e`, I prefer to correctly trap the error in my bash script. But in that case (bash script inside a perl script inside openQA) this make sense.